### PR TITLE
Wrapped token login bug

### PR DESCRIPTION
**Pull request by zofskeez**
As reported in #17355, visiting the logout endpoint with a `wrapped_token` was not automatically logging in if the user was already logged out. The transition to the logout route was not being caught by the `EXCLUDED_REDIRECT_URLS` whitelist because of the query parameter. After the token was unwrapped and auth was successful, the user would be redirected back to logout and then the auth route where the token would again be unwrapped which would fail the second time.

wrapped-token

I also discovered while testing this that the wrapped token can be passed to the auth route and it will automatically log in as well.

wrapped-token2

### CODE REVIEW

File: ui/app/mixins/cluster-route.js [hunk 1]
```diff
@@ -29,11 +29,13 @@ export default Mixin.create({
       targetRoute !== transition.targetName &&
       targetRoute !== this.router.currentRouteName
     ) {
+      // there may be query params so check for inclusion rather than exact match
```
Comment (hellobontempo) [inline ui/app/mixins/cluster-route.js:32]: thanks for the comment!

File: ui/tests/acceptance/wrapped-token-test.js [hunk 2]
```diff
@@ -41,4 +43,13 @@ module('Acceptance | wrapped_token query param functionality', function (hooks)
     await settled();
     assert.strictEqual(currentURL(), '/vault/secrets', 'authenticates and redirects to home');
   });
+
+  test('it should authenticate when hitting logout url with wrapped_token when logged out', async function (assert) {
```
Comment (hellobontempo) [inline ui/tests/acceptance/wrapped-token-test.js:47]: thank you for adding a test!

**Review by hellobontempo — APPROVED** [review]
Nice job getting this bug fix in!

**Comment by remilapeyre** [comment]
As far as I am aware this feature is not documented. Could fixing the bug be the opportunity to add some documentation for this or should it be done in another PR?

**Comment by zofskeez** [comment]
> As far as I am aware this feature is not documented. Could fixing the bug be the opportunity to add some documentation for this or should it be done in another PR?

Yes that's a good suggestion. I have forwarded the request along and it will be handled in another PR.

**Pull request merged**